include/cuco/detail/static_map/static_map_ref.inl

```diff
@@ -577,7 +577,7 @@ class operator_impl<
    */
 
   template <typename Value, typename Op>
-  __device__ void insert_or_apply(Value const& value, Op op)
+  __device__ bool insert_or_apply(Value const& value, Op op)
```

Comment:
This could be a separate PR but I'm fine with keeping it in here since the change is so small.

Reply:
Also, we have to document the return value in our Doxygen headers

Reply:
Is there any contraint that the device API signature should be same as host API signature ?

Reply:
Nah, host and device APIs have different signatures in most cases